### PR TITLE
Bugfix for #17492: Do not prepend PWD when path is in form user@server:path or server:path

### LINKED ISSUE #17492: synchronize 'pull' mode incorrectly prepends PWD to src
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

plugins/action/synchronize.py
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = 
  configured module search path = Default w/o overrides

```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

'pull' mode preprends PWD to the (remote) src path, e.g.

/home/blah/blah/rsync://some-host/path/on/remote
##### STEPS TO REPRODUCE

Try the example in the synchronize docs:

```
synchronize: mode=pull src=rsync://somehost.com/path/ dest=/some/absolute/path/

```

##### EXPECTED RESULTS

rsync from somehost.com/path/
##### ACTUAL RESULTS

failed rsync from ${PWD}rsync://somehost.com/path/

```

```

The problem seems to be around the dest_is_local flag. If dest_is_local (true for pull with remote src), the 'rsync://' prefix is not handled.

### PR DESCRIPTION
**Pull request by pneerincx**
##### SUMMARY
This fixes #17492. There are several instances where the _get_absolute_path function is called. This should only happen for relative paths on a local file system and hence
 * not for paths on a remote server
 * nor for absolute paths.
In some cases the code in the synchronize.py plugin would check for presence of a colon (path on remote server) and if the path starts with a slash forward (absolute path). In two cases the code would only check for absolute paths, but failed to check for remote paths. This results in ${PWD} getting prepended to the remote path leading to something like /path/to/PWD/user@server:/path/to/what/needs/to/be/synced. 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
synchronize plugin

##### ANSIBLE VERSION
```
ansible 2.4.2.0
```

### CODE REVIEW

**Comment by pneerincx** [comment]
ready_for_review

**Comment by pneerincx** [comment]
bot_status

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,7 @@ rpm-build
 # Eclipse/PyDev stuff...
 .project
 .pydevproject
+.settings
```
Comment (mattclay) [inline .gitignore:16]: This change should be made in a separate PR since it's not directly related to the rest of the changes here.

**Comment by mkrizek** [comment]
Thank you very much for your interest in Ansible. This plugin is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/ansible.posix.

Please re-submit this issue in the above repository.

If you have further questions please stop by IRC or the mailing list:

* IRC: #ansible on irc.freenode.net
* mailing list: https://groups.google.com/forum/#!forum/ansible-project
